gms/gossiper.cc

```diff
@@ -762,16 +762,38 @@ future<> gossiper::failure_detector_loop_for_node(gms::inet_address node, int64_
     auto diff = gossiper::clk::duration(0);
     auto echo_interval = std::chrono::milliseconds(2000);
     auto max_duration = echo_interval + std::chrono::milliseconds(_failure_detector_timeout_ms());
+    // We're using an abort_source*
+    // so we can use a new one on each iteration
+    abort_source* asp = nullptr;
+    auto tmr = timer([&asp] {
+        // asp must be engaged when the timer is armed
+        asp->request_abort_ex(timed_out_error());
+    });
+    auto sub = _abort_source.subscribe([&tmr, &asp] () noexcept {
+        tmr.cancel();
+        // The timer may have already fired
+        // so protect against double abort
+        if (asp && !asp->abort_requested()) {
+            asp->request_abort_ex(abort_requested_exception());
+        }
+    });
```

Comment:
This isn't the intent of abort_source. It's meant to be a long-term object that manages shutdowns, not something you create for every request.

Reply:
does it really matter?
it's so light weight that it can be used efficiently per request, certainly on the control plain.